service/collector.go

```diff
@@ -172,6 +172,11 @@ LOOP:
 	for {
 		select {
 		case err := <-col.set.ConfigProvider.Watch():
+			if !col.set.EnableConfigWatch {
```

Comment:
I thought more about this feature, and found that if we have `EnableConfigWatch` a config in the `ConfigProviderSettings`, then send a "nil" watcher that does disables "watching" completely.

What do you think?

Reply:
I thought about it, and ended up putting this in the collector loop for two reasons:
* It's more visible and explicit this way. If it was part of `ConfigProvider` logic, then we'd have a case in the collector loop that implicitly wouldn't do anything depending on settings.
* It's possible for distributions to provide their own `ConfigProvider` implementations, and in that case we'd be forcing them to implement this logic on their own.

That said, I don't think it really matters *that* much where it is.

Reply:
I would go this way for the moment... Feels more natural to be there... Happy to hear other opinions.

Reply:
I'm with @bogdandrutu here. Disabling (or not enabling) the config watch should not have the file watched at all. I understand the sentiment, but I also prefer to not have this code executed at all unless the config watch is enabled.

Reply:
@swiatekm-sumo we are blocked on this :)

Reply:
I had misunderstood your original comment @bogdandrutu, and thought you wanted to leave this as is. Sorry about that, I'm fine moving this check to a lower level.

If we want to do no additional work at all if this is disabled, then we'd need to pass this value down the the Resolver and indicate to the provider we don't want notifications by passing a nil watcher function here: https://github.com/open-telemetry/opentelemetry-collector/blob/ae077bec51f81e838bacbb522f5711636f7a83e0/confmap/resolver.go#L312. Does that sound reasonable?